pkg/node/statestore.go

```diff
@@ -87,3 +96,30 @@ func overlayNonceExists(s storage.StateStorer) ([]byte, bool, error) {
 func setOverlayNonce(s storage.StateStorer, overlayNonce []byte) error {
 	return s.Put(OverlayNonce, overlayNonce)
 }
+
+func migrateStamperData(stateStore storage.StateStorer, stamperStore storage.Store) error {
```

Comment:
this will run everytime the node starts up, right?
Maybe this should be part of some migration logic that runs only once after upgrading to the new release?

Reply:
The idea was that after the migration, future runs would be noop since the stamps would have been migrated to stamperstore and deleted from statestore. Then we can remove the code later on.

How can we tie this migration to the new release?

Reply:
https://github.com/ethersphere/bee/pull/4154 takes care of this, right @aloknerurkar ?

Reply:
@istae So right now the problem is the stampissuer data is in statestore. But going ahead it will be in stamper store. This is why the statestore migration doesnt remove these entries. Instead this separate migration is required. This migration will happen after the statestore migration. So the entries will be ported to new statestore then moved to stamper store and deleted.

Only the first time it starts there will be postage related entries in the statestore. So the migration will be a no-op after that.

Reply:
Agree with @istae Is there a way this can be done as part of the migration logic?

Reply:
I dont think so, the migration logic is localized to the store itself. Here we need to take the entries from a completely different store. Which is why this has to be specific I think.